feat: add preliminary Windows support and mruby_dir feature (#32)
genya0407/minutus: README.md

You can also use mruby's `master` branch, but it is not tested.

> If you need to customize a specific version (such as a specific tag), you can do so via `mruby_dir`. See "Local Dir" for details.

Reviewer: [nits]

```suggestion
If you need to use mruby versions not listed above (e.g., a specific tag or a fork), you can do so via `mruby_dir` feature. See "mruby_dir" section for details.
```

minutus = { version = "*", features = ["mruby_master"] }
```

### Local Dir

> Use a local directory instead of a remote Git repository.

Reviewer: [nits]

```suggestion
### mruby_dir

When `mruby_dir` feature is specified, minutus uses your local directory as `mruby`.
```

```toml
# Cargo.toml
#
[dependencies]
minutus = { version = "*", features = ["mruby_dir"] }
```

> 

Reviewer: We had better describe `MINUTUS_MRUBY_DIR` here, like:

````suggestion
```sh
# POSIX-sh
#
env MINUTUS_MRUBY_DIR=/path/to/my/mruby cargo build
```
````

Author: After some consideration, I decided to configure the environment variables in the **Rakefile** instead of using POSIX sh. 

```ruby
# The value of `MINUTUS_MRUBY_DIR` is the path to the source code of mruby.
ENV["minutus_mruby_dir".upcase] = "/tmp/mruby-stable"
```

I chose to use `'minutus_mruby_dir'.upcase` rather than `MINUTUS_MRUBY_DIR`, as I personally find it more readable and expressive.

It might be better to update minutus-mrbgem-template directly, rather than relying on users to follow the instructions in the README manually.

@genya0407 As the maintainer of this project, you can improve it yourself later.

Reviewer: Writing the environment variable in Rakefile in Example section is good.
But, I want to describe `MINUTUS_MRUBY_DIR` **before** example section.

Reason 1:
Generally speaking, I would like to provide a complete description before the example.
I think examples are meant to help the user understand the feature or describe additional uses, and so on.

Reason 2:
minutus has two usecases:

1. Embed mruby code in Rust code
    * https://github.com/genya0407/minutus/tree/trunk?tab=readme-ov-file#embed-mruby-in-rust
2. Create mrbgem by Rust
    * https://github.com/genya0407/minutus/tree/trunk?tab=readme-ov-file#create-mrbgem-by-rust

`mruby_dir` is for both usecases, however, in former one, users do not necessarily have `Rakefile`.

---

If you do not have strong opinions about the README, could I rewrite it after merging this PR?
After my rewrite, if you happen to have any concerns about the README, please feel free to propose any changes.

Of course, if you want to continue discussion, that's welcome!

Author: I have no objections to this.

#### Example

```shell
# POSIX-sh
#
cd /tmp
tag="240412a29080f775fabf97e56fc158b7249367e1"
curl -Lfo mruby.tgz "https://github.com/mruby/mruby/archive/${tag}.tar.gz"

tar -xf mruby.tgz
mv mruby-$tag mruby-local-dir
```

**build:**

Modify your project's `Rakefile`.

```ruby
# Rakefile
#
file :mruby do
  #sh "git clone --depth=1 https://github.com/mruby/mruby.git"
  local_dir = "/tmp/mruby-local-dir"
  unless Dir.exist?("mruby")
    FileUtils.cp_r(local_dir, "mruby")
  end
  # ...
```

Next, you can pass the specified path using the **MINUTUS_MRUBY_DIR** environment variable.

```sh
# POSIX-sh
#

> env MINUTUS_MRUBY_DIR=/tmp/mruby-local-dir cargo b -rvv

Reviewer: Why not use the "canonical form" of command? (`cargo build`, not `cargo b`)
And I think we shouldn't add extra options unless it's necessary to avoid confusion for users.

```suggestion
env MINUTUS_MRUBY_DIR=/tmp/mruby-local-dir cargo build
```

Author: Sorry, that was my oversight. 
When I wrote the README, I didn’t think too much about it—it just came out naturally.

You can use `cargo --list` to see all available Cargo subcommands. Although `b` is a built-in alias provided by Cargo, I agree that `build` is more intuitive. 

The output of `cargo build -h` includes options like `-r, --release` and `-v, --verbose... Use verbose output (-vv very verbose/build.rs output)`. 

While `-rvv` might not be immediately intuitive, it's not hard to understand.
The reason for using `-vv` is to capture the output from `build.rs` during the build process, since errors can occur at that stage. Giving users more visibility into the build process is not necessarily a bad thing.

```

## Supported Platforms

**minutus** primarily supports Unix-like systems.

> If you encounter issues in a Windows MSVC environment, try using **msys2** and the `*-pc-windows-gnu` target.

Reviewer: 👍
